internal/docker/docker.go

```diff
@@ -58,6 +91,72 @@ func CheckProcessStateOnTheHost(containerName string, process string, state stri
 	return nil
 }
 
+// CopyFileToContainer copies a file to the running container
+func CopyFileToContainer(ctx context.Context, containerName string, srcPath string, parentDir string, isTar bool) error {
+	dockerClient := getDockerClient()
+
+	log.WithFields(log.Fields{
+		"container": containerName,
+		"src":       srcPath,
+		"parentDir": parentDir,
+	}).Trace("Copying file to container")
+
+	targetDirectory := filepath.Dir(parentDir)
+
+	_, err := dockerClient.ContainerStatPath(ctx, containerName, targetDirectory)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"container": containerName,
+			"error":     err,
+			"src":       srcPath,
+			"target":    targetDirectory,
+		}).Error("Could not get parent directory in the container")
+		return err
+	}
+
+	file, err := os.Open(srcPath)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"container": containerName,
+			"error":     err,
+			"src":       srcPath,
+			"parentDir": parentDir,
+		}).Error("Could not open file to deploy")
+		return err
+	}
+	defer file.Close()
+
+	// TODO: detect the file has TAR headers
```

Comment:
Ive used this github.com/gabriel-vasile/mimetype with a lot of success for this sort of thing

Reply:
Let's do this in a follow-up PR